proparse/src/main/java/org/prorefactor/proparse/antlr4/PreproEval.java

```diff
@@ -0,0 +1,612 @@
+
+    // Match any number of chars
+    if (b.charAt(0) == '*') {
+      return matches(a, b.substring(1)) || matches(a.substring(1), b);
+    }
+
+    // Match an escaped char
+    if (b.charAt(0) == '~') {
+      return a.charAt(0) == b.charAt(1) && matches(a.substring(1), b.substring(2));
+    }
+
+    // Match a single specific char
+    return a.charAt(0) == b.charAt(0) && matches(a.substring(1), b.substring(1));
+  }
+
+  static Integer numentries(Object a, Object b) {
+    String sa = getString(a);
+    if (sa.length() == 0)
+      return 0;
+    String sb;
+    if (b != null) {
+      sb = getString(b);
+    } else {
+      sb = ",";
+    }
+    Pattern regex = Pattern.compile(sb, Pattern.LITERAL | Pattern.CASE_INSENSITIVE);
+    return regex.split(sa, -2).length;
+  }
+
+  static Object opDivide(Object left, Object right) {
+    if (left == null || right == null)
+      return null;
+    if ((left instanceof Integer) && (right instanceof Integer))
+      return (Integer) left / (Integer) right;
+    if ((left instanceof Number) && (right instanceof Number)) {
+      Double fl = ((Number) left).doubleValue();
+      Double fr = ((Number) right).doubleValue();
+      return fl / fr;
+    }
+    throw new ProEvalException("Incompatible data type in expression.");
+  }
+
+  static Object opMinus(Object left, Object right) {
+    if (left == null || right == null)
+      return null;
+    if ((left instanceof Integer) && (right instanceof Integer))
+      return (Integer) left - (Integer) right;
+    if ((left instanceof Number) && (right instanceof Number)) {
+      Double fl = ((Number) left).doubleValue();
+      Double fr = ((Number) right).doubleValue();
+      return fl - fr;
+    }
+    throw new ProEvalException("Incompatible data type in expression.");
+  }
+
+  static Object opMultiply(Object left, Object right) {
+    if (left == null || right == null)
+      return null;
+    if ((left instanceof Integer) && (right instanceof Integer))
+      return (Integer) left * (Integer) right;
+    if ((left instanceof Number) && (right instanceof Number)) {
+      Double fl = ((Number) left).doubleValue();
+      Double fr = ((Number) right).doubleValue();
+      return fl * fr;
+    }
+    throw new ProEvalException("Incompatible data type in expression.");
+  }
+
+  static Object opPlus(Object left, Object right) {
+    if (left == null || right == null)
+      return null;
+    if ((left instanceof String) && (right instanceof String))
+      return (String) left + right;
+    if ((left instanceof Integer) && (right instanceof Integer))
+      return (Integer) left + (Integer) right;
+    if ((left instanceof Number) && (right instanceof Number)) {
+      Double fl = ((Number) left).doubleValue();
+      Double fr = ((Number) right).doubleValue();
+      return fl + fr;
+    }
+    throw new ProEvalException("Incompatible data type in expression.");
+  }
+
+  // TODO Verify if this not a duplicate of settings#getPropath()
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1135)